### ISSUE TITLE
Allow specific files to be served

### ISSUE THREAD

**Issue by mikkotikkanen** [expressjs/serve-static#40]
It would be nice if I could, in addition to serve files from fe. `serveStatic("public/")`, I could also serve specific files from other directories fe.

```js
serveStatic("js/", ["node_modules/bootstrap/dist/js/bootstrap.js"]);
serveStatic("css/", ["node_modules/bootstrap/dist/css/bootstrap.css"]);
```

..so one could fe. just glob the files that he needs to be shared to clientside, pass them to `serve-static` and call it a day.

**Comment by dougwilson** [comment]
Gotcha. I mean, I believe you can currently do this today using, for example:
```js
app.use('/js/bootstrap.js', serveStatic('node_modules/bootstrap/dist/js/bootstrap.js'))
```

But if not, or that don't work for you, let me know. And PRs are welcome for new functionality, if it is desired to happen quickly :)

**Comment by dougwilson** [comment]
Gotcha. I mean, I believe you can currently do this today using, for example:
```js
app.all('/js/bootstrap.js', serveStatic('node_modules/bootstrap/dist/'))
```

But if not, or that don't work for you, let me know. And PRs are welcome for new functionality, if it is desired to happen quickly :)

**Comment by mikkotikkanen** [comment]
The example in question works, however, that is going to result conflicts, thus I would like to serve the file from another directory, and just changing the mounting point doesn't seem to work. :/

```js
app.all('/quickstart/bootstrap', serveStatic('node_modules/bootstrap/dist'));
```

**Comment by dougwilson** [comment]
Gotcha. Well, please feel free to put together a PR :) otherwise I don't really promise any kind of timeline on this.

**Comment by dougwilson** [comment]
Hi @mikkotikkanen, I'm going to close this since you haven't made a pull request, and neither has anyone else. I re-read this and just don't understand what you're trying to do. Looking at your file https://github.com/mikkotikkanen/express-quickstart/blob/master/index.js you are trying to do something you can already do with this module, just I think you didn't understand (or I didn't understand your question).

For example, the following:
```js
app.all('/quickstart/jquery/js/*', function(req, res, next) {
  send(req, req.params[0], {root: 'node_modules/express-quickstart/node_modules/jquery/dist'})
  .on('error', function(err) { return next(); })
  .pipe(res);
});
app.all('/quickstart/bootstrap/*', function(req, res, next) {
  send(req, req.params[0], {root: 'node_modules/express-quickstart/node_modules/bootstrap/dist'})
  .on('error', function(err) { return next(); })
  .pipe(res);
});
app.all('/quickstart/font-awesome/css/*', function(req, res, next) {
  send(req, req.params[0], {root: 'node_modules/express-quickstart/node_modules/font-awesome/css'})
  .on('error', function(err) { return next(); })
  .pipe(res);
});
app.all('/quickstart/font-awesome/fonts/*', function(req, res, next) {
  send(req, req.params[0], {root: 'node_modules/express-quickstart/node_modules/font-awesome/fonts'})
  .on('error', function(err) { return next(); })
  .pipe(res);
});
app.all('/quickstart/requirejs/js/require.js', function(req, res, next) {
  send(req, 'require.js', {root: 'node_modules/express-quickstart/node_modules/requirejs'})
  .on('error', function(err) { return next(); })
  .pipe(res);
});
```

Can be written as the following:
```js
app.use('/quickstart/jquery/js/', serveStatic('node_modules/express-quickstart/node_modules/jquery/dist'));
app.use('/quickstart/bootstrap/', serveStatic('node_modules/express-quickstart/node_modules/bootstrap/dist'));
app.use('/quickstart/font-awesome/css/', serveStatic('node_modules/express-quickstart/node_modules/font-awesome/css'));
app.use('/quickstart/font-awesome/fonts/', serveStatic('node_modules/express-quickstart/node_modules/font-awesome/fonts'));
app.use('/quickstart/requirejs/js/', serveStatic('node_modules/express-quickstart/node_modules/requirejs'));
```

**Comment by mikkotikkanen** [comment]
Get Outlook for Android






On Tue, Jun 7, 2016 at 4:20 AM +0300, "Douglas Wilson" <notifications@github.com> wrote:












Closed #40.
